.github/workflows/docs.yml

```
@@ -0,0 +1,25 @@
+name: Docs
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-22.04
+    container:
+      image: crystallang/crystal
+    steps:
+      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608
+      - name: Build docs
+        run: crystal docs --project-name Representer --project-version 1.2.0-Dev
+      - name: Deploy docs
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+        uses: JamesIves/github-pages-deploy-action@v4
```

Could you pin this to a commit SHA?

```suggestion
        uses: JamesIves/github-pages-deploy-action@v4
```

---

I dont understand what you mean

---

Don't use `@v4` but use `@<commit sha>`. See https://exercism.org/docs/building/github/gha-best-practices#h-pin-actions-to-shas